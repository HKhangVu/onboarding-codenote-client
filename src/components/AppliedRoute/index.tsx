import React from 'react';
import { Route } from 'react-router-dom';

interface INewProps {
  component: any,
  props?:any,
  [id:string]:any,
}

const AppliedRoute = ({ component: C, props: cProps, ...rest }:INewProps) => {
  return (
    <Route {...rest} 
    render={props => {
      return (
        <C {...props} {...cProps} />
      );
    }} 
    />
  );
}
  

export default (AppliedRoute);
