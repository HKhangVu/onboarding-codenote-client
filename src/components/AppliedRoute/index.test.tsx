import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux';
import Home from '../../screens/Home';
import Navigation from '../Navigation';
import store from '../../store';
import AppliedRoute from './index';

describe("test render AppliedRoute", () => {
  test('renders AppliedRoute have all params', () => {
    let rest: { [index: string]: string } = {}
    rest['path'] = "/";
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter  >
          <AppliedRoute component={Home} props={"string"} {...rest} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  test('renders AppliedRoute do not have props', () => {
    let rest: { [index: string]: { message: string } } = {}
    rest['a'] = { message: 'some message' };
    const wrapper = shallow(<AppliedRoute component={Home} {...rest} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders AppliedRoute Home do not have rest', () => {
    const wrapper = shallow(<AppliedRoute component={Home} props={new Object()} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders AppliedRoute Navigation only have component', () => {
    const wrapper = shallow(<AppliedRoute component={Navigation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
