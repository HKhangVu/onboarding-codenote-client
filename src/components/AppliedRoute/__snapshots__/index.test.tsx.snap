// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render AppliedRoute renders AppliedRoute Home do not have rest 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`test render AppliedRoute renders AppliedRoute Navigation only have component 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`test render AppliedRoute renders AppliedRoute do not have props 1`] = `
<Route
  a={
    Object {
      "message": "some message",
    }
  }
  render={[Function]}
/>
`;
